### PR TITLE
feat: flesh out dashboard

### PR DESCRIPTION
**Pull request by LoneRifle**
image

## Problem and Solution

Flesh out the interface for the dashboard so that we can at least see the list of checkers available to the user

- Insert a 5-across grid into Projects page
- Define a common theme for cards to be displayed on grid
- Implement CreateNew and CheckerCard layout per wireframe specs

- Rename projects to dashboard, to avoid overlap of terms (project v checker)
- On Dashboard mounting, retrieve the list of checkers from `/api/v1/c` and render them as cards onto the dashboard

- Invoke a Modal from CreateNew, prompting the user to enter id, title and description
- On successful submit, reload the dashboard

### CODE REVIEW

File: src/client/components/dashboard/CreateNewModal.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  return (
+    <Modal isOpen={isOpen} onClose={resetAndClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>
+          {checker ? `Copy ${checker.id}` : 'Create New Checker'}
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <Input
```
Comment (lamkeewei) [inline src/client/components/dashboard/CreateNewModal.tsx:113]: Minor point, maybe want to consider using [react-hook-form](https://react-hook-form.com/) instead to save on  boilerplate? It's already added for the main checker form. 

```jsx
const { register, handleSubmit, errors, formState } = useForm()
const { isValid, isValidating } = formState

// Submit returns { id: ..., title: ... }
const onSubmit = (data) => console.log(data);

// ...

<form onSubmit={handleSubmit(onSubmit)}>
  <FormControl isInvalid={errors.title}>
     <Input name="title" ref={register({ required: true })} />
     <FormErrorMessage>{errors.title && errors.title.message}</FormErrorMessage>
  </FormControl>

  <Button disabled={!isValid} type="submit">Create</Button>
</form>
```

Comment (LoneRifle) [reply src/client/components/dashboard/CreateNewModal.tsx:113]: I'll look into this

File: src/client/pages/Dashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { Container, Flex, Grid, VStack } from '@chakra-ui/react'
+
+import { ApiClient } from '../api'
+import { Navbar, CreateNew, CheckerCard } from '../components/dashboard'
+import { Checker } from '../../types/checker'
+
+export const Dashboard: FC = () => {
+  const [checkers, setCheckers] = useState<Checker[]>([])
+
+  const loadCheckers = async () => {
```
Comment (lamkeewei) [inline src/client/pages/Dashboard.tsx:11]: Sorry for introducing so many libs...but [react-query](https://react-query.tanstack.com/quick-start) can help here. Provides out-the-box loading/error status. 

```ts
const { isLoading, isError, data: checker } = useQuery('checkers', () => ApiClient.get('/c))

// Render spinner whenever fetching...
{ isLoading && <Spinner />}
```

Then in other components that that needs to refetch checkers, you can use [query invalidation](https://react-query.tanstack.com/guides/query-invalidation):
```ts
const queryClient = useQueryClient()
queryClient.invalidate('checkers')
```

**Review by lamkeewei — APPROVED** [review]
lgtm!

**Pull request merged**